### PR TITLE
wait for virtual endpoint gateway to be available after creation

### PR DESCRIPTION
**Pull request by deepaksibm**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'

...
```

### CODE REVIEW

**Comment by deepaksibm** [comment]
```

$ make testacc TEST=./ibm/service/vpc TESTARGS='-run=TestAccIBMISVirtualEndpointGateway_Basic'
=== RUN   TestAccIBMISVirtualEndpointGateway_Basic
--- PASS: TestAccIBMISVirtualEndpointGateway_Basic (97.86s)
=== RUN   TestAccIBMISVirtualEndpointGateway_Basic_SecurityGroups
--- PASS: TestAccIBMISVirtualEndpointGateway_Basic_SecurityGroups (93.95s)
PASS
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/vpc     193.526s
```

**Comment by deepaksibm** [comment]
Closes #2236

File: ibm/service/vpc/resource_ibm_is_virtual_endpoint_gateway.go [hunk 1]
```diff
@@ -307,6 +307,10 @@ func resourceIBMisVirtualEndpointGatewayCreate(d *schema.ResourceData, meta inte
 	}
 
 	d.SetId(*result.ID)
+	_, err = isWaitForVirtualEndpointGatewayAvailable(sess, d.Id(), d.Timeout(schema.TimeoutUpdate))
```
Comment (hkantare) [inline ibm/service/vpc/resource_ibm_is_virtual_endpoint_gateway.go:310]: should we use schema.TimeoutCreate?

Comment (deepaksibm) [reply ibm/service/vpc/resource_ibm_is_virtual_endpoint_gateway.go:310]: https://github.com/IBM-Cloud/terraform-provider-ibm/pull/4206/files#diff-a0d9b3493bb9618e40ba9b89e9cd9f93b54b8ba4de246bf6f6701cbfbbf72f53R62
Timeouts: &schema.ResourceTimeout is already defined

Comment (hkantare) [reply ibm/service/vpc/resource_ibm_is_virtual_endpoint_gateway.go:310]: why we have used update we shd use create right

Comment (deepaksibm) [reply ibm/service/vpc/resource_ibm_is_virtual_endpoint_gateway.go]: updated

**Pull request merged**